### DISCUSSION TITLE
How to configure multi language with versioning enabled

### DISCUSSION THREAD

**Discussion by sukki37** [squidfunk/mkdocs-material#3646] [Q&A]
I have encountered some difficulties when using mkdocs-material:
1. When I tried to enable the multi-language feature,  I found it can not set a default language and there is no documentation on that.
2. I need some instruction on how to configure multi-language with versioning enabled.

**Comment by squidfunk** [comment]
A very good guide on setting up a multi-language build: #2346

Versioning goes on top of that. You basically build a project with multiple `mkdocs.yml` in one repository which you then version with `mike` as described in [the versioning guide](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/). There shouldn't be much more to it. You might also look into the [mkdocs-static-i18n](https://github.com/ultrabug/mkdocs-static-i18n) plugin, which supports a default language as I understand.

**Reply by sukki37** [reply]
Thank you for your prompt reply. I have read the introduction #2346,  but I'm still a little confused. In #2346,  koen-yesplan said that:
> One small disadvantage is that you cannot use the mkdocs serve command to serve your site locally with all three languages. MkDocs can only serve one language at a time.

So the question is that there is no mkdoc.yml in the root directory, so I don't know how/where can I exec `mike deploy --push --update-aliases` command...

**Reply by squidfunk** [reply]
Yes, I guess the mkdocs-static-i18n plugin together with `mike` might be a better choice in that case.

**Reply by sukki37** [reply]
Thank you for your recommendation. I've tried already,  but unfortunately, this plugin couldn't support multi-level nav translation for now. In my case, maybe I will create a new repository for our Chinese site and just set `alternative.link` to an absolute URL to achieve our goal.

Hope that `mkdocs-material` can improve the ability for integrating multi-version and multi-language support.

Thanks again.
